pyuvdata/uvcal/uvcal.py

```diff
@@ -1264,6 +1682,374 @@ def _check_freq_spacing(self, *, raise_errors=True):
+            if temp_id is not None:
+                if cat_diffs == 0:
+                    # Everything matches, return the catalog ID of the matching entry
+                    return temp_id
+                warnings.warn(
+                    f"The provided name {cat_name} is already used but has different "
+                    "parameters. Adding another entry with the same name but a "
+                    "different ID and parameters."
+                )
+
+        # If source is unique, begin creating a dictionary for it
+        self.phase_center_catalog[cat_id] = cat_entry
+        self.Nphase = len(self.phase_center_catalog)
+
+        return cat_id
+
+    def _remove_phase_center(self, defunct_id):
+        """
+        Remove an entry from the internal object/source catalog.
+
+        Removes an entry from the attribute `phase_center_catalog`.
+
+        Parameters
+        ----------
+        defunct_id : int
+            Catalog ID of the source to be removed
+
+        Raises
+        ------
+        IndexError
+            If the name provided is not found as a key in `phase_center_catalog`
+        """
+        if defunct_id not in self.phase_center_catalog:
+            raise IndexError("No source by that ID contained in the catalog.")
+
+        del self.phase_center_catalog[defunct_id]
+        self.Nphase = len(self.phase_center_catalog)
+
+    def _clear_unused_phase_centers(self):
+        """
+        Remove objects dictionaries and names that are no longer in use.
+
+        Goes through the `phase_center_catalog` attribute in of a UVData object and
+        clears out entries that are no longer being used, and appropriately updates
+        `phase_center_id_array` accordingly. This function is not typically called
+        by users, but instead is used by other methods.
+
+        """
+        unique_cat_ids = np.unique(self.phase_center_id_array)
+        defunct_list = []
+        Nphase = 0
+        for cat_id in self.phase_center_catalog:
+            if cat_id in unique_cat_ids:
+                Nphase += 1
+            else:
+                defunct_list.append(cat_id)
+
+        # Check the number of "good" sources we have -- if we haven't dropped any,
+        # then we are free to bail, otherwise update the Nphase attribute
+        if Nphase == self.Nphase:
+            return
+
+        # Time to kill the entries that are no longer in the source stack
+        for defunct_id in defunct_list:
+            self._remove_phase_center(defunct_id)
+
+    def print_phase_center_info(
+        self,
+        catalog_identifier=None,
+        *,
+        hms_format=None,
+        return_str=False,
+        print_table=True,
+    ):
+        """
+        Print out the details of the phase centers.
+
+        Prints out an ASCII table that contains the details of the
+        `phase_center_catalog` attribute, which acts as the internal source catalog
+        for UVData objects.
```

Comment:
UVData -> UVCal

Reply:
Fixed in 766e32646768181b70ddd8d2e576d90ca57c254c.